gpii/node_modules/gpii-iniFile/index.js

```diff
@@ -0,0 +1,454 @@
+
+            if (handlers.gotValue) {
+                var quoted = match.value_ml_quote !== undefined || match.value_quote !== undefined;
+                var newValue = handlers.gotValue(sectionPath, match.key, value, quoted);
+                if (options.write) {
+                    if (newValue === fluid.NO_VALUE) {
+                        result = "";
+                    } else if (newValue !== undefined) {
+                        var quote = (match.qqq || match.q || "");
+                        if (match.value_ml_indent) {
+                            newValue = newValue.replace(/\n/g, "\n" + match.indent + " ".repeat(4));
+                        }
+                        result = match.prefix + quote + newValue + quote + match.suffix;
+                    }
+                }
+            }
+        }
+
+        return result;
+    });
+
+    var append = endSection(0);
+    if (options.write && append) {
+        output += append;
+    }
+
+    return options.write && output;
+};
+
+/**
+ * Writes an object to existing INI file content.
+ *
+ * @param {String} input The ini file content to update (can be empty).
+ * @param {Object} data The new settings data.
+ * @param {Object} options Options, mostly for writing new values.
+ * @param {String} options.keyValueDelimiter - Text which separates keys and values, for new values. [default: "="].
+ * @param {Boolean} options.keepUndefined - true to keep items and sections that are not in data. Use to define values
+ * without reading the file beforehand.
+ * @param {String} options.multilineStyle - How new multi-line values are written:
+ *   "'''" or '"""': Surround the value with 3x single or double quotes (default).
+ *   "indent": Indent the additional lines.
+ *   "escape": Use an escaped n (\n).
+ *   anything else: Wrap the value with the given value.
+ * @param {String} options.quote - For new values, quote them "always", "never", "strings" (for only strings), or
+ *  "spaces" (if the value starts or ends with a space) [default: "spaces"].
+ * @param {String} options.quoteChar - The quote character for new values when quoting. [default: " (double quote)].
+ * @param {String} options.eol - The file's end of line character(s). [default: auto-detect]
+ *
+ * @return {String} The new ini file content.
+ */
+gpii.iniFile.write = function (input, data, options) {
+    options = Object.assign({
+        keyValueDelimiter: "=",
+        keepUndefined: false,
+        eol: undefined,
+        quoteChar: "\"",
+        quote: "spaces",
+        multilineStyle: "\"\"\""
+    }, options);
+
+    if (options.eol === undefined) {
+        var m = /\r\n|[\r\n]/.exec(input);
+        options.eol = m && m[0] || os.EOL;
+    }
+
+    options.write = true;
+
+    /**
+     * Stringify a value - converts it to a string as returned by JSON.stringify.
+     * @param {Mixed} value The value.
+     * @return {String} The stringified version of value.
+     */
+    var stringify = function (value) {
```

Comment:
Please stick this nested material (stringify, writeValue, handlers) etc. onto globals to aid patching